rhizome/host/lib/vm_setup.rb

```diff
@@ -445,6 +445,7 @@ def cloudinit(unix_user, public_keys, gua, nics, swap_size_bytes, boot_image)
 listen-address=#{dnsmasq_address_ip6}
 dhcp-option=26,1400
 bind-interfaces
+address=/localhost.blob.core.windows.net/#{nics.first.net4.split("/").first}
```

Comment:
whoa though, this affects every VM right? Can't you use /etc/hosts as a GHA special to do something similar?

Reply:
Yes, it affects every VM. We've discussed with Furkan whether it is safe for VMs and opened in that way but I can  check to do that for runner vms only. I can't use /etc/hosts as I need to have that mapping inside containers as well. I need to have that for containers because docker layer cache uses buildkit to build images and it caches files via container.

Reply:
Upon further discussion, we think that we can get away by checking the boot_image so that it doesn't impact every VM.

Reply:
Oof, containers, yes...I guess this the most well-sealed way to do it.

Reply:
With https://github.com/ubicloud/ubicloud/pull/2016 we are assigning ipv4 address to dnsmasq so it can used for dns resolution. I've also updated the code to have that address mapping only for github vms.